custom-elements/reactions/resources/reactions.js

```diff
@@ -78,6 +78,38 @@ function testNodeDisconnector(testFunction, name) {
     container.parentNode.removeChild(container);
 }
 
+function testCloner(testFunction, name) {
+    let container = document.createElement('div');
+    container.appendChild(document.createElement('div'));
+    document.body.appendChild(container);
+
+    test(function () {
+        var element = defineNewCustomElement(['id']);
+        var instance = document.createElement(element.name);
+        container.appendChild(instance);
+
+        instance.setAttribute('id', 'foo');
+        assert_array_equals(element.log().types(), ['constructed', 'connected', 'attributeChanged']);
+        var newInstance = testFunction(instance);
+        var logEntries = element.log();
+        assert_array_equals(logEntries.types(), ['constructed', 'attributeChanged']);
+        assert_equals(logEntries.log().name, 'id');
+        assert_equals(logEntries.log().oldValue, null);
+        assert_equals(logEntries.log().newValue, 'foo');
+    }, name + ' must enqueue a attributeChanged reaction when cloning an element with an observed attribute');
+
+    test(function () {
+        var element = defineNewCustomElement(['id']);
+        var instance = document.createElement(element.name);
+        container.appendChild(instance);
+
+        instance.setAttribute('lang', 'en');
+        assert_array_equals(element.log().types(), ['constructed', 'connected']);
+        var newInstance = testFunction(instance);
+        assert_array_equals(element.log().types(), ['constructed']);
+    }, name + ' must not enqueue a attributeChanged reaction when cloning an element with an unobserved attribute');
```

Comment:
Sure, I added a new test case. But I think in general, those edge cases are better tested in https://github.com/w3c/web-platform-tests/pull/3853 which targets each reaction type more throughly.  These individual CEReaction tests are just sanity check for each IDL attribute so I don't think we necessary have to exhaustively test each edge case.